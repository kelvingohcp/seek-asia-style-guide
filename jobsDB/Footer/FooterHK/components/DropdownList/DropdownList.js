import styles from './DropdownList.less';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Text, ChevronIcon } from 'seek-asia-style-guide/react';

export default class DropdownList extends Component {
  constructor() {
    super();

    this.state = {
      chevronDirection: 'down',
      isDropdownOpen: false
    };
  }

  toggleDropdown() {
    this.setState({
      chevronDirection: (this.state.chevronDirection === 'down') ? 'up' : 'down',
      isDropdownOpen: !this.state.isDropdownOpen
    });
  }

  render() {
    const { value, compact, noShadow, descriptionProps, className, children } = this.props;
    return (
      <div
        className={classnames({
          [styles.root]: true,
          [styles.noShadow]: noShadow,
          [className]: className
        })} >
        <div className={styles.listItemValue} onClick={this.toggleDropdown.bind(this)} >
          <Text
            loud={!compact}
            waving={compact}
            baseline={false}
            className={styles.chevronRight}
            raw={true}
            {...descriptionProps}>
            {value}
          </Text>
          <ChevronIcon className={styles.chevronFixHeight} direction={this.state.chevronDirection} />
        </div>
        <div className={this.state.isDropdownOpen ? styles.dropdown : styles.dropdownNoShow}>
          {children}
        </div>
      </div>
    );
  }
}

DropdownList.propTypes = {
  value: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  compact: PropTypes.bool,
  noShadow: PropTypes.bool,
  descriptionProps: PropTypes.object
};

DropdownList.defaultProps = {
  compact: false,
  noShadow: false
};
