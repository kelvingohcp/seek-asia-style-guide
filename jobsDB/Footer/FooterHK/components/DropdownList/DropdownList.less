@import (reference) "~seek-asia-style-guide/theme";
@import "~seek-asia-style-guide/react/ListItem/ListItem.less";

@chevron-padding-left: @grid-base * 2;
@dropdown-padding-top: @grid-base * 2;
@chevron-height: @grid-base * 4;

.root {
  flex-direction: column;
  background-color: inherit;
  color: @saGrey2;
  padding-left: inherit;
  padding-bottom: inherit;
}

.chevronFixHeight {
  height: @chevron-height;
}

.chevronRight {
  padding-right: @chevron-padding-left;
}

.listItemValue {
  display: flex;
  cursor: pointer;
}

.dropdown {
  display: flex;
  flex-direction: column;
  padding-top: @dropdown-padding-top;
}

.dropdownNoShow:extend(.dropdown) {
  display: none;
}
