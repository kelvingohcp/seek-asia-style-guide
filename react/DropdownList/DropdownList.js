import styles from './DropdownList.less';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Text, ChevronIcon, ListItem, Button } from 'seek-asia-style-guide/react';

export default class DropdownList extends Component {
  constructor() {
    super();

    this.state = {
      chevronDirection: 'down',
      isDropdownOpen: false
    };
  }

  toggleDropdown() {
    this.setState({
      chevronDirection: (this.state.chevronDirection === 'down') ? 'up' : 'down',
      isDropdownOpen: !this.state.isDropdownOpen
    });
  }

  render() {
    const { value, noShadow, className, children, compact, ...restProps } = this.props;

    const listItemProps = { noShadow: true };
    const dropdownListProps = { noShadow: true, className: styles.lowerLevel };
    const buttonProps = { className: styles.button };
    if (compact) {
      listItemProps.compact = compact;
      buttonProps.compact = compact;
      dropdownListProps.compact = compact;
    }

    return (
      <div
        className={classnames({
          [styles.root]: true,
          [styles.noShadow]: noShadow,
          [styles.expanded]: this.state.isDropdownOpen,
          [className]: className
        })} >
        <div className={styles.listItemValue} onClick={this.toggleDropdown.bind(this)} >
          <Text
            shouting={!compact}
            baseline={false}
            className={styles.chevronRight}
            raw={true}
            {...restProps}>
            {value}
          </Text>
          <ChevronIcon className={styles.chevronFixHeight} direction={this.state.chevronDirection} />
        </div>
        <div className={this.state.isDropdownOpen ? styles.dropdown : styles.dropdownNoShow}>
          {
            React.Children.map(children, child => {
              switch (child.type) {
                case ListItem: return React.cloneElement(child, listItemProps);
                case DropdownList: return React.cloneElement(child, dropdownListProps);
                case Button: return React.cloneElement(child, buttonProps);
                default: return child;
              }
            })
          }
        </div>
      </div>
    );
  }
}

DropdownList.propTypes = {
  value: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  noShadow: PropTypes.bool,
  compact: PropTypes.bool
};

DropdownList.defaultProps = {
  noShadow: false,
  compact: false
};
