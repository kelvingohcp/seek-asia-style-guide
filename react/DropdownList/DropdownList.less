@import (reference) "~seek-asia-style-guide/theme";
@import (reference) "~seek-asia-style-guide/react/ListItem/ListItem.less";

@chevron-padding-left: @grid-base * 2;
@dropdown-padding-top: @grid-base * 2;
@chevron-height: @grid-base * 4;
@chevron-padding-top: @grid-base;
@button-padding: @grid-base * 4;

.root {
  align-items: flex-start;
  box-shadow: 0 1px 1px @saGrey5;
  
  color: @saGrey1;
  display: flex;
  line-height: @list-item-line-height;
  padding: @list-item-h-padding @list-item-v-padding;
  flex-direction: column;
  &.lowerLevel.expanded {
    padding-bottom: 0;
  }
}

.noShadow {
  box-shadow: none;
}

.button {
  outline: none;
  line-height: normal;
  height: auto;
  white-space: normal;
  padding: @button-padding;
  text-align: left;
}

.chevronFixHeight {
  height: @chevron-height;
  margin-top: @chevron-padding-top;
}

.chevronRight {
  padding-right: @chevron-padding-left;
}

.listItemValue {
  display: flex;
  cursor: pointer;
}

.dropdown {
  display: flex;
  flex-direction: column;
  padding-top: @dropdown-padding-top;
}

.dropdownNoShow:extend(.dropdown) {
  display: none;
}
