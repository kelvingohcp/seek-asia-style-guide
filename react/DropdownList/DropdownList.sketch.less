.root {
  width: fit-content;
}

.button {
  line-height: normal;
  height: auto;
  white-space: normal;
  text-align: left;
}
