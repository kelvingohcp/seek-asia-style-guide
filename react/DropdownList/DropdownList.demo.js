import React from 'react';
import DropdownList from './DropdownList';
import * as sketch from './DropdownList.sketch';
import { ListItem } from 'seek-asia-style-guide/react';

const list = [
  <ListItem key="1" value="Children 1" />,
  <ListItem key="2" value="Children 2" />,
  <ListItem key="3" value="Children 3" />,
  <ListItem key="4" value="Children 4" />
];

export default {
  route: '/DropdownList',
  title: 'Dropdown List',
  component: DropdownList,
  sketch,
  initialProps: {
    value: 'Category',
    children: list
  },
  options: [
    {
      label: 'Compact',
      type: 'checklist',
      states: [
        {
          label: 'Compact',
          transformProps: props => ({
            ...props,
            compact: true
          })
        }
      ]
    },
    {
      label: 'No shadow',
      type: 'checklist',
      states: [
        {
          label: 'No shadow',
          transformProps: props => ({
            ...props,
            noShadow: true
          })
        }
      ]
    },
    {
      label: 'Description Text Weight',
      type: 'radio',
      states: [
        {
          label: 'Regular',
          transformProps: props => props
        },
        {
          label: 'Strong',
          transformProps: props => ({
            ...props,
            strong: true
          })
        },
        {
          label: 'Light',
          transformProps: props => ({
            ...props,
            light: true
          })
        },
        {
          label: 'Semi-strong',
          transformProps: props => ({
            ...props,
            semiStrong: true
          })
        }
      ]
    },
    {
      label: 'Description Text Color',
      type: 'radio',
      states: [
        {
          label: 'Neutral tone',
          transformProps: props => props
        },
        {
          label: 'Positive',
          transformProps: props => ({
            ...props,
            positive: true
          })
        },
        {
          label: 'Critical',
          transformProps: props => ({
            ...props,
            critical: true
          })
        },
        {
          label: 'Secondary',
          transformProps: props => ({
            ...props,
            secondary: true
          })
        }
      ]
    }
  ]
};
