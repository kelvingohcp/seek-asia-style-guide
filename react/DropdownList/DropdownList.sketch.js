import React from 'react';
import DropdownList from './DropdownList';
import mapKeys from 'lodash/mapKeys';
import styles from './DropdownList.sketch.less';
import { ListItem, Button } from 'seek-asia-style-guide/react';

export const dropdownList = {
  'value, children': (
    <DropdownList value='Description' className={styles.root}>
      <ListItem value="Children" />
    </DropdownList>
  ),
  'value, children, no shadow': (
    <DropdownList value='Description' className={styles.root} noShadow>
      <ListItem value="Children" />
    </DropdownList>
  ),
  'value, children, compact': (
    <DropdownList value='Description' className={styles.root} compact>
      <ListItem value="Children" />
    </DropdownList>
  ),
  'value, nested children': (
    <DropdownList value='Description' className={styles.root}>
      <ListItem value="Children 1" />
      <Button color="hyperlink">
        Children 2
      </Button>
      <DropdownList value="Sub-category" compact>
        <ListItem value="Children 3" />
      </DropdownList>
    </DropdownList>
  ),
  'value, nested children, long wordings': (
    <DropdownList value='At jobsDB, we believe in an open and inclusive world for all. Here, opportunities and possibilities are plenty: A fresh start. A new career. A welcome change. These are just a few of the things we help people explore.' className={styles.root}>
      <ListItem value="As a leading job portal with substantial positions across China, Hong Kong, Indonesia, Philippines, Singapore and Thailand, we are Asia’s preferred destination for job seekers. With a team of 1,000 talented individuals, we’ve a strong local presence in every country we’re in." />
      <Button color="hyperlink">
        With our understanding and insights on each market, as well our passion for technology, we provide job seekers the best opportunities and deliver exceptional value to employers.
      </Button>
      <DropdownList value="Sub-category" compact>
        <ListItem value="By really understanding our job seekers and employers, we ensure they find what’s right for them. After all, we’re not just about filling vacancies but fulfilling true potential." />
      </DropdownList>
    </DropdownList>
  )
};

// Export text styles as symbols
// (at least until Sketch libraries support shared text styles)
export const symbols = mapKeys(dropdownList, (value, key) => `DropdownList/${key}`);
