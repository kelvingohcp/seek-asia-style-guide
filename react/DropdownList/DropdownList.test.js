import React from 'react';
import { shallow } from 'enzyme';
import DropdownList from './DropdownList';
import styles from './DropdownList.less';

describe('DropdownList:', () => {
  it('should render value with children', () => {
    const wrapper = shallow(
      <DropdownList value="testValue">
        <span />
      </DropdownList>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with center-aligned chevron', () => {
    const wrapper = shallow(
      <DropdownList value="testValue" chevronAlignment="center">
        <span />
      </DropdownList>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without shadow', () => {
    const wrapper = shallow(
      <DropdownList value="testValue" noShadow>
        <span />
      </DropdownList>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render as lower level child', () => {
    const wrapper = shallow(
      <DropdownList value="testValue">
        <DropdownList value="testValue" lowerLevel>
          <span />
        </DropdownList>
      </DropdownList>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render in compact mode', () => {
    const wrapper = shallow(
      <DropdownList value="testValue" compact>
        <span />
      </DropdownList>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with className', () => {
    const wrapper = shallow(
      <DropdownList value="testValue" className={styles.root}>
        <span />
      </DropdownList>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
