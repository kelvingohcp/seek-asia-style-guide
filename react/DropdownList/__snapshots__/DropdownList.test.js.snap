// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownList: should render in compact mode 1`] = `
<div
  className="root"
>
  <div
    className="listItemValue"
    onClick={[Function]}
  >
    <Text
      baseline={false}
      className="chevronRight"
      raw={true}
      shouting={false}
    >
      testValue
    </Text>
    <ChevronIcon
      className="chevronFixHeight"
      direction="down"
    />
  </div>
  <div
    className="dropdownNoShow"
  >
    <span
      key=".0"
    />
  </div>
</div>
`;

exports[`DropdownList: should render value with children 1`] = `
<div
  className="root"
>
  <div
    className="listItemValue"
    onClick={[Function]}
  >
    <Text
      baseline={false}
      className="chevronRight"
      raw={true}
      shouting={true}
    >
      testValue
    </Text>
    <ChevronIcon
      className="chevronFixHeight"
      direction="down"
    />
  </div>
  <div
    className="dropdownNoShow"
  >
    <span
      key=".0"
    />
  </div>
</div>
`;

exports[`DropdownList: should render with className 1`] = `
<div
  className="root"
>
  <div
    className="listItemValue"
    onClick={[Function]}
  >
    <Text
      baseline={false}
      className="chevronRight"
      raw={true}
      shouting={true}
    >
      testValue
    </Text>
    <ChevronIcon
      className="chevronFixHeight"
      direction="down"
    />
  </div>
  <div
    className="dropdownNoShow"
  >
    <span
      key=".0"
    />
  </div>
</div>
`;

exports[`DropdownList: should render without shadow 1`] = `
<div
  className="root noShadow"
>
  <div
    className="listItemValue"
    onClick={[Function]}
  >
    <Text
      baseline={false}
      className="chevronRight"
      raw={true}
      shouting={true}
    >
      testValue
    </Text>
    <ChevronIcon
      className="chevronFixHeight"
      direction="down"
    />
  </div>
  <div
    className="dropdownNoShow"
  >
    <span
      key=".0"
    />
  </div>
</div>
`;
